src/Workspaces/Core/Portable/Execution/Asset.cs

```diff
@@ -121,7 +121,7 @@ internal sealed class AnalyzerReferenceAsset : Asset
         private readonly AnalyzerReference _reference;
 
         public AnalyzerReferenceAsset(Serializer serializer, AnalyzerReference reference, Checksum checksum, string kind) :
-            base(checksum, WellKnownChecksumObjects.AnalyzerReference)
+            base(Checksum.Create(kind, checksum), WellKnownChecksumObjects.AnalyzerReference)
         {
             Contract.Requires(kind == WellKnownChecksumObjects.AnalyzerReference);
```

Comment:
this method is give to a delegate which has a set signature. so every method has same signature. that is why it has kind in it.